### PR TITLE
Initial v1beta2 (experimental) cut

### PR DESCRIPTION
**Pull request by smarterclayton**
@ncdc review please

This enables v1beta2 as experimental, exposing only the *new* v1beta2 objects.

### CODE REVIEW

**Comment by smarterclayton** [comment]
@bparees @pweil- @ironcladlou this does not include builds, routes, or deployments, the latter two will need a "spec" / "status", and the former should include the issues discussed in other threads.  Can you guys make sure there is a card / issue to track each of those sections into v1beta2?  Thanks.

**Comment by smarterclayton** [comment]
@deads2k @liggitt see the last commit - in order to properly default values (using the scheme defaults) the internal version needs to go through a conversion.  This makes defaults versionable (which they aren't today).

**Comment by deads2k** [comment]
I'm not clear why we would want to version our default values is a way that is separate from us building the config itself.  I would have said that the `buildSerializeableConfig` methods were a giant defaulter intended for matched set usage.  Either we'd serialize out however we want into a fresh file or we'd use the memory copy directly.  I can't envision a scenario where we'd want the config to built in a stale scheme.

**Comment by smarterclayton** [comment]
If you have value defaulting, you can't change those defaults once real values are in the wild.  Since we will ship config that we have to add things to, we will have to set default values for a particular version of API objects.  So defaults have to come from a version.  It's really no different than our other API objects.

**Comment by deads2k** [comment]
Defaults that are set during read will happen during the existing conversion process.  If it's a concern that there will be value drift, why not continue to build the config using the in-memory representation and round-trip it through conversion before usage?

**Comment by deads2k** [comment]
Wouldn't we always round trip through the latest version of the scheme?  I don't see why we'd continue using the old scheme for building the config in code.

It seems like our config validation (against the in-memory package) is going to force any bad defaulting into the open.  At the very least, it would prevent usage.

**Comment by smarterclayton** [comment]
So that we don't have support to round tripping old config versions.

**Comment by smarterclayton** [comment]
The only risk is if we have behavior that can't be expressed in the external storage version (like objects internally that are read).  If you think we have no expectation of doing that, we can round trip to latest.

However, it's looking likely that we will start using versioned objects internally for all the reasons mentioned and a few more.  Validation is likely to move to external objects as well.  We don't have to beat the rush, but we do need to be aware.

File: pkg/image/api/v1beta2/types.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	// Status describes the current state of this stream
+	Status ImageStreamStatus `json:"status,omitempty"`
+}
+
+// ImageStreamSpec represents options for ImageStreams.
+type ImageStreamSpec struct {
+	// Optional, if specified this stream is backed by a Docker repository on this server
+	DockerImageRepository string `json:"dockerImageRepository,omitempty"`
+	// Tags map arbitrary string values to specific image locators
+	Tags []NamedTagReference `json:"tags,omitempty"`
```
Comment (ncdc) [inline pkg/image/api/v1beta2/types.go]: I know we just created this as part of the ImageStream work, but I'm questioning if TagReference is the appropriate name for this type. It's entirely valid to have annotations for a tag without there being any "reference" part. Would TagSpec or something along those lines be more appropriate?

Comment (smarterclayton) [inline pkg/image/api/v1beta2/types.go]: Sure.

----- Original Message -----

**Comment by pweil-** [comment]
> @bparees @pweil- @ironcladlou this does not include builds, routes, or deployments, the latter two will need a "spec" / "status", and the former should include the issues discussed in other threads. Can you guys make sure there is a card / issue to track each of those sections into v1beta2? Thanks.

Yep, I have a card for the multi-port changes that were in the new rebase

**Comment by smarterclayton** [comment]
@deads2k all comments addressed PTAL

**Comment by deads2k** [comment]
One comment, otherwise the last commit lgtm.

**Comment by derekwaynecarr** [comment]
[test]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**